app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/ActionsBottomSheetDialog.kt

```diff
@@ -17,134 +17,24 @@
-
-        setArchiveUi()
-
-        archive.setClosingOnClickListener { onClickListener.onArchive() }
-        markAsReadUnread.setClosingOnClickListener { onClickListener.onReadUnread() }
-        move.setClosingOnClickListener { onClickListener.onMove() }
-        postpone.setClosingOnClickListener { onClickListener.onPostpone() }
-        favorite.setClosingOnClickListener { onClickListener.onFavorite() }
-        spam.setClosingOnClickListener { onClickListener.onSpam() }
-        reportJunk.setClosingOnClickListener { onClickListener.onReportJunk() }
-        print.setClosingOnClickListener { onClickListener.onPrint() }
-        reportDisplayProblem.setClosingOnClickListener { onClickListener.onReportDisplayProblem() }
-
-        mainActions.setClosingOnClickListener { id: Int ->
-            when (id) {
-                R.id.actionReply -> onClickListener.onReply()
-                R.id.actionReplyAll -> onClickListener.onReplyAll()
-                R.id.actionForward -> onClickListener.onForward()
-                R.id.actionDelete -> onClickListener.onDelete()
-            }
-        }
-    }
-
-    fun initOnClickListener(listener: OnActionClick) {
-        onClickListener = listener
-    }
-
-    private fun computeUnreadStyle(isSeen: Boolean) = if (isSeen) {
-        R.drawable.ic_envelope to R.string.actionMarkAsUnread
-    } else {
-        R.drawable.ic_envelope_open to R.string.actionMarkAsRead
-    }
-
-    private fun computeFavoriteStyle(isFavorite: Boolean): Pair<Int, Int> {
-        return if (isFavorite) R.drawable.ic_unstar to R.string.actionUnstar else R.drawable.ic_star to R.string.actionStar
-    }
-
-    fun setMarkAsReadUi(isSeen: Boolean) = with(binding.markAsReadUnread) {
-        val (readIconRes, readTextRes) = computeUnreadStyle(isSeen)
-        setIconResource(readIconRes)
-        setText(readTextRes)
-    }
-
-    fun setFavoriteUi(isFavorite: Boolean) = with(binding.favorite) {
-        val (favoriteIconRes, favoriteText) = computeFavoriteStyle(isFavorite)
-        setIconResource(favoriteIconRes)
-        setText(favoriteText)
-    }
-
-    private fun setArchiveUi() = with(binding.archive) {
-        if (mainViewModel.isCurrentFolderRole(FolderRole.ARCHIVE)) {
-            setIconResource(R.drawable.ic_drawer_inbox)
-            setText(R.string.actionMoveToInbox)
-        }
-    }
-
-    private fun ActionItemView.setClosingOnClickListener(callback: (() -> Unit)) {
+    protected fun ActionItemView.setClosingOnClickListener(callback: () -> Unit) {
         setOnClickListener {
             findNavController().popBackStack()
             callback()
         }
     }
 
-    private fun MainActionsView.setClosingOnClickListener(callback: ((Int) -> Unit)) {
+    protected fun MainActionsView.setClosingOnClickListener(callback: (Int) -> Unit) {
         setOnItemClickListener { id ->
-            callback(id)
             findNavController().popBackStack()
+            callback(id)
         }
     }
 }
```

Comment:
Are we sure that's the correct order for all situations ?

By order, I mean `callback()` then `popBackStack()`, or `popBackStack()` then `callback()`.